### ISSUE TITLE
Add mixins by new proposal

### ISSUE THREAD

**Issue by ai** [MoOx/postcss-cssnext#203]
New proposal by @tabatkins with mixins by `@apply` rule http://tabatkins.github.io/specs/css-apply-rule/

**Comment by MoOx** [comment]
This one require postcss v5, will add that after #202

https://github.com/pascalduez/postcss-apply

**Comment by keithamus** [comment]
For any who land here, it looks as though #208 is the new issue that needs to be resolved before this lands.

**Comment by Jcan** [comment]
Hey MoOx - we are very excited about the function. do you have a timeline for it? Thanks ;-)

**Comment by Jcan** [comment]
I think Mixin is very useful. This is a simple example:

:root {
    --clearfix: {
        content: '';
        display: table;
        clear: both;
    }
}

#header {
    ......
    &::after: var(--clearfix);
}

--
Look forward for the new feature!

**Comment by MoOx** [comment]
If someone can double check that a plugin exists and follow the spec, this should be easy to add (PR welcome, and btw, here is the way to add a feature (maybe outdated a little, but should help, it's very easy to add a new plugin)  https://github.com/MoOx/postcss-cssnext/blob/master/CONTRIBUTING.md#add-a-feature)

**Comment by zeke** [comment]
Not sure I have the time or the chops to take this on, but would love to see it supported in cssnext!

**Comment by pascalduez** [comment]
The aim of [`postcss-apply`](https://github.com/pascalduez/postcss-apply) is to follow the "spec" as close as possible, but I feel it's a bit early to name it a "spec"?
Also it's suffering the same limitations as for custom properties, no cascading or inheritance so far.
I get to make it more clear on the readme.

Anyhow, I would be happy to see it used in CSSnext.

**Comment by MoOx** [comment]
As far as the limitation does not produce any change with or without the transformation, it's not an issue. postcss-custom-properties supported code will works in modern browsers without any different behavior.

I didn't read the spec at all so I can't say anything for now (and to be honest, I am not personally interested yet by this feature, and may never be - that don't mean it does not have its place to this project).

**Comment by pascalduez** [comment]
Just wanted to show good willing after the last days "social highlight".
I guess it should live alongside postcss-custom-properties and follow the same guidelines, since they are pretty close.
But postcss-apply is just a handfull LOC after all, no big deal.

Open to move it to the PostCSS org if you feel it's needed.

Collaboration over fragmentation.

**Comment by MoOx** [comment]
@pascalduez feel free to make a PR with proper doc about limitation (like custom props).

**Comment by MoOx** [comment]
If anyone want to handle this, it should take a minute if the corresponding postcss plugin is ready.
See the doc to add a feature https://github.com/MoOx/postcss-cssnext/blob/master/CONTRIBUTING.md#add-a-feature

**Comment by pascalduez** [comment]
@MoOx I'll give it a spin, sorry for the delay. 
I tried using the plugins together, there might be a few tweaks needed up the pipeline.
